eventstore/whatsapp_actions.py

```diff
@@ -40,3 +40,32 @@ def update_rapidpro_preferred_channel(message):
     update_rapidpro_contact.delay(
         urn=f"whatsapp:{message.contact_id}", fields={"preferred_channel": "WhatsApp"}
     )
+
+
+def handle_event(event):
+    """
+    Triggers all the actions that are required for this event
+    """
+    if not event.fallback_channel:
+        handle_whatsapp_event(event)
+
+
+def handle_whatsapp_event(event):
+    # 410: Message expired
+    if any(error["code"] == 410 for error in event.data["errors"]):
+        # TODO: handle whatsapp timeout system event
+        pass
+    elif settings.ENABLE_UNSENT_EVENT_ACTION:
+        hsm_error = False
+        for error in event.data["errors"]:
+            if "structure unavailable" in error["title"]:
+                hsm_error = True
+            if "envelope mismatch" in error["title"]:
+                hsm_error = True
+
+        if hsm_error:
+            async_create_flow_start.delay(
+                extra={},
+                flow=settings.RAPIDPRO_UNSENT_EVENT_FLOW,
+                urns=[f"whatsapp:{event.recipient_id}"],
+            )
```

Comment:
I'm wondering if this logic should be inside the `Event` model instance, so that we can do an `if event.is_hsm_error: ...` in here.

Reply:
I like this idea, would make it less complex. I'm going to implement